Nos/Views/New Note/ImagePickerUIViewController.swift

```diff
@@ -48,9 +49,30 @@ struct ImagePickerUIViewController: UIViewControllerRepresentable {
                 onCompletion(videoURL)
             } else if let imageURL = info[.imageURL] as? URL {
                 onCompletion(imageURL)
+            } else if let image = info[.originalImage] as? UIImage,
+                let imageData = image.jpegData(compressionQuality: 1.0) {
+                let url = saveImage(imageData)
+                onCompletion(url)
             } else {
                 onCompletion(nil)
             }
         }
+
+        /// Saves the given image data to a JPG file and returns the URL of the file.
+        /// - Parameter imageData: The image data to save to disk.
+        /// - Returns: The URL of the image file.
+        private func saveImage(_ imageData: Data) -> URL? {
+            let urls = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)
+            let documentDirectory = urls[0]
+            let fileURL = documentDirectory.appendingPathComponent("capturedImage.jpg")
+
+            do {
+                try imageData.write(to: fileURL, options: .atomic)
+                return fileURL
+            } catch {
```

Comment:
Catching errors here means that saveImage doesn't throw errors and onCompletion(url) above is then called with a valid url without realizing the photo the user took couldn't be saved. Shouldn't we mark this function with throws so that at least onCompletion is called with a nil value (to be completionists, we should be presenting an error dialog instead, so onCompletion should receive a Result object).

Reply:
I think I see what you're getting at here, though what happens isn't exactly what you've described. If we catch an error, this func returns `nil`, so `onCompletion` is called with `nil`. As this was a bug fix, I didn't even consider refactoring the signature of `onCompletion` -- I certainly could, but that requires a bit more testing to make sure we don't introduce a regression.

Reply:
Sounds about right. Good work!